### PR TITLE
Switch CI to GitHub Actions

### PR DESCRIPTION
**Pull request by pwnall**

### CODE REVIEW

**Comment by pwnall** [comment]
@cmumford PTAL?

The changes to `CMakeLists.txt` are the same as in #54 -- I intend to eliminate them by rebasing after #54 lands.

**Comment by pwnall** [comment]
I intend to remove the AppVeyor integration after this PR gets approved, so the AppVeyor failures here are OK.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      run: >-
+        cmake -S "${{ github.workspace }}" -B "${{ env.CMAKE_BUILD_DIR }}"
+        -DCMAKE_BUILD_TYPE=${{ env.CMAKE_BUILD_TYPE }}
+        -DCMAKE_INSTALL_PREFIX=${{ runner.temp }}/install_test/
+        -DBUILD_SHARED_LIBS=${{ matrix.shared_lib && '1' || '0' }}
+        -DCRC32C_USE_GLOG=${{ matrix.use_glog && '1' || '0' }}
+
+    - name: Build
+      run: >-
+        cmake --build "${{ env.CMAKE_BUILD_DIR }}"
```
[the remark this answers is not in the document]
Comment (pwnall) [reply .github/workflows/build.yml:89]: WDYT about tackling this as a follow-up optimization?

The README and [the code](https://github.com/SimenB/github-actions-cpu-cores/blob/main/index.js) suggest that the action exports the core count to the GitHub Actions environment. The count still needs to be plugged into various tools (`npx` in the README example).

The Travis CI config uses `ninja` (which uses all cores by default) on Linux and macOS. The AppVeyor config uses MSVC on Windows. The GitHub Actions config only regresses parallelism on Linux and macOS, and I think we'd be looking at saving 20-30 seconds on these OSes -- GitHub's hosted runners are really fast.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+    - name: Run C API Tests on Unix
+      if: ${{ runner.os != 'Windows' }}
+      run: >-
+        ${{ github.workspace }}/build/crc32c_capi_tests
+
+    - name: Run C API Tests on Windows
+      if: ${{ runner.os == 'Windows' }}
+      run: >-
+        ${{ github.workspace }}\build\${{ env.CMAKE_BUILD_TYPE }}\crc32c_capi_tests.exe
```
[the remark this answers is not in the document]
Comment (pwnall) [reply .github/workflows/build.yml]: I removed the step duplication. WDYT?

I'd rather not use `ctest` -- we started out with it, and it misbehaved (hangs, eating output) when tests crashed. I switched to running the test binaries directly to work around these issues. We could re-investigate and revisit after the initial CI switch.

File: .github/workflows/build.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (cmumford) [reply .github/workflows/build.yml:89]: SGTM.

File: .github/workflows/build.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (cmumford) [reply .github/workflows/build.yml]: SGTM.

**Comment by pwnall** [comment]
@cmumford  Thank you very much for the quick review!

**Pull request merged**